dxtbx/model/scaling_model.py

```diff
@@ -0,0 +1,72 @@
+from __future__ import absolute_import, division
+
+from abc import ABCMeta, abstractmethod
+
+class ScalingModelBaseIface(object):
+  '''
+  A basic scaling model class. This assumes no interface other than
+  serialization methods. Classes inheriting from this class can
+  then be serialized via a simple registry
+
+  '''
+
+  __metaclass__ = ABCMeta
+
+  @abstractmethod
+  def to_dict(self):
+    '''
+    Convert the scaling model to dictionary
+
+    '''
+    pass
+
+  @abstractmethod
+  def from_dict(cls, obj):
+    '''
+    Convert a dictionary to the scaling model
+
+    '''
+    pass
+
+
+class ScalingModelFactory(object):
```

Comment:
Is this class cargo-cul... erm, copied from somewhere, or is this from scratch?
In any case would suggest using the python setuptools registry rather than rolling our own.

Reply:
This should probably be done for the profile model code as well at some point as well!